experiment/Commonfunctions/commonfunctions-demo.js

```diff
@@ -38,21 +45,24 @@ function startNext(){
         else{bft();}
             flag_started=1;
     }
-    else{console.log("next");
+    else{
+
+  
         if(i==ans.length)return;
         visitElement(ans[i],animX1);
         d3.select("#traversal").append("text").text(ans[i].name + " ");
-        i++;
+        changei(i+1);
```

Comment:
changei() basically just changes the value of I to the value in the bracket. Since I am modularizing the code I ran across an error where svg,root and i couldn't be assigned or changed unless you passed them through a function. Also as mentioned above I don't know what i does still so I had to name that as the function.